woocommerce/woocommerce-ios
# Custom Currency - Mark III

In this PR, we'll pull together the MoneyFormatterSettings, MoneyFormatter, and Currency structs.

The architecture goes like this:
- `Currency` is the model for retrieving bits of information. It's where you can look up a currency symbol or find the user's preferred currency option settings.
- `CurrencySettings` is where the user's preferred currency options are stored. This class listens for changes made to the settings.
- `CurrencyFormatter` is the main utility class where amounts can be converted from string to decimal and where amounts are formatted to fit the user's currency display preferences.
- `MoneyFormatter` is the old class that will soon be chopped out and replaced.

@bummytime I have no idea what I'm doing as far as architecture. Stab in the dark!!!

## To test
- Check that all unit tests are green
- Do all the names make sense?
- Does the architecture make sense?
